### PR TITLE
Change default warn color to yellow

### PR DESCRIPTION
**Pull request by KristofferC**
Change the warning color to what I believe is a better default (to yellow from red).

Reasons
- Red is a bit harsh for a warning.
- Googling around for warning messages it seems that the vast majority uses yellow for warnings. Following what appears to be a standard makes sense.
- Having a visual difference between warnings and errors seems sensible.

xterm default
image

gnome-terminal default (ubuntu default)
image

gnome-terminal white background (this one is not so nice...)
image

powershell
image

win cmd
image

cc @twadleigh

### CODE REVIEW

**Comment by tkelman** [comment]
guess the only question is why was it set to red from the beginning?

**Comment by KristofferC** [comment]
Seem to have been there from the start https://github.com/JuliaLang/julia/commit/8bbd04ec5b4b192eee52dfa3e73c102bde97f1be#diff-c20eb2eead14027fab4293e10bacf73bR443 3 years ago?

**Comment by KristofferC** [comment]
Hah! Found it https://github.com/JuliaLang/julia/commit/68f63db0dccb561bc606138d317950267d6d4785#diff-2b2ebeef35e91b1fff15d49dc1d526b5R50. I tgot changed from yellow to red 3 ½ years ago.

@StefanKarpinski didn't like the yellow.

**Comment by ViralBShah** [comment]
Yes, Red for errors, yellow for warning goes nicely with traffic light colors.

**Comment by nalimilan** [comment]
Definitely a good idea to differentiate warnings from errors. I've had students who thought (in R) that a package wasn't installed correctly just because an innocent warning was printed in red.

**Comment by TotalVerb** [comment]
To make everyone happy, can we use 256-colour orange if it's available, and fall back to yellow otherwise? Or is this possibly infeasible for portability?

**Comment by StefanKarpinski** [comment]
I don't really have an issue with yellow for warnings. This is why we really should ship our own terminal of some kind on Windows. On UNIX we're always going to need to deal with people's terminals, but xterm-265 should be possible on most UNIX systems.

**Comment by okvs** [comment]
My impression is that most hardcore SW developers use dark terminal/editor backgrounds. But I hope you will at least give some consideration to users who prefer light backgrounds. Yellow on white (and also cyan on white) are not great combinations. I certainly understand that it is hard, bordering on impossible, to choose colours that look good for both light and dark backgrounds.

I also have a (non-empirical, gut-level) hunch that people who use light backgrounds are less likely to tweak their setup.

**Comment by KristofferC** [comment]
Yellow on white is not so bad. The problem is that we are printing everything in bold all the time:

image

**Comment by okvs** [comment]
Actually, I see that bold yellow looks ok for me, but the gnome-terminal version was bad. And honestly, I don't really think the colour below is yellow, it's more like beige.

screen shot 2016-09-13 at 09 53 05

**Comment by KristofferC** [comment]
Well, it is defined to be yellow at least.. http://pueblo.sourceforge.net/doc/manual/ansi_color_codes.html

**Comment by KristofferC** [comment]
> In version 2.50 and later, bold is interpreted as ‘high-intensity’, which was the original ANSI definition. (High-intensity colors are brighter than low-intensity ones, which are the default.)

I guess some terminals does that which is why the bold yellow looks so bad on white.

**Comment by KristofferC** [comment]
@TotalVerb AFAIU, detecting if a terminal supports 256 colors is tricky because terminals lie through their teeth. I can however give you https://github.com/JuliaLang/julia/pull/18473 so you can set orange yourself.

**Comment by ViralBShah** [comment]
I kinda prefer orange over yellow.

**Comment by KristofferC** [comment]
And after #18473 you can easily set that with an env var. What we are talking about here is the default color and for that orange is out of the question due to not being supported in many default / unconfigured terminals.

**Comment by KristofferC** [comment]
Added news entry, anyone opposed of merging?

**Comment by kshyatt** [comment]
Bump.

**Comment by KristofferC** [comment]
From the 9 thumbs up, 0 thumbs down and the triviality of reverting, I will merge this at the end of the weekend if no one has objections.

**Comment by tkelman** [comment]
That yellow will be pretty bad on light terminals. Is determining the background color any more doable than detecting 256 color support? We could also do yellow on Windows and orange elsewhere since I get the impression far fewer people use light background terminals on Windows, and it's the only place where 256 color support isn't a safe assumption.

**Comment by KristofferC** [comment]
Guess I will focus on the non boldness in the colorized messages then which is what is causing this to look bad on white backgrounds.

**Comment by KristofferC** [comment]
Rebased + hooked up pkg + test to use the warn color function.

I tried the online merge resolution tool and it sort of messed up my branch... Anyway the total diff of these commit seems ok so should be fine to squash.

**Review by tkelman — APPROVED** [review]
so now this prints the lead "WARNING" in bold but the rest of the content not bold, right?

File: base/test.jl [hunk 1]
```diff
@@ -140,7 +140,7 @@ type Broken <: Result
     orig_expr
 end
 function Base.show(io::IO, t::Broken)
-    print_with_color(:yellow, io, "Test Broken\n"; bold = true)
+    print_with_color(:Base.warn_color(), io, "Test Broken\n"; bold = true)
```
Comment (MichaelHatherly) [inline base/test.jl]: Is the `:` meant to be prefixed to `Base` here, or just a typo?

Comment (KristofferC) [inline base/test.jl]: typo (which I thought I fixed but maybe I forgot to commit the file).

**Comment by KristofferC** [comment]
Yes @tkelman.

**Review by Sacha0 — APPROVED** [review]

File: NEWS.md [hunk 2]
```diff
@@ -60,9 +60,9 @@ Library improvements
     Therefore, light versions of the colors are now supported.
     For the available colors see the help entry on `print_with_color`.
 
-  * The default color for info messages has been changed from blue to cyan ([#18442]).
```
Comment (tkelman) [inline NEWS.md]: we should put this link back (and add 18453 too)

**Pull request merged**